File: java/src/main/java/dev/formhandle/cli/Prompt.java
package dev.formhandle.cli;

import java.util.Scanner;

public class Prompt {
    private static final Scanner scanner = new Scanner(System.in);

    public static String ask(String question) {
        System.out.print(question);
        System.out.flush();
        if (!scanner.hasNextLine()) {
            System.out.println();
            System.exit(1);
        }
        return scanner.nextLine().trim();
    }

    public static boolean confirm(String question) {
        String answer = ask(question + " (y/N) ");
        return answer.equalsIgnoreCase("y") || answer.equalsIgnoreCase("yes");
    }
}

File: php/src/Prompt.php
<?php

declare(strict_types=1);

namespace FormHandle;

class Prompt
{
    public static function ask(string $question): string
    {
        echo $question;
        $line = fgets(STDIN);
        if ($line === false) {
            echo "\n";
            exit(1);
        }
        return trim($line);
    }

    public static function confirm(string $question): bool
    {
        $answer = self::ask("$question (y/N) ");
        return in_array(strtolower($answer), ['y', 'yes'], true);
    }
}
